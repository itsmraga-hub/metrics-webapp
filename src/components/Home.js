import { useSelector } from 'react-redux';

import { useState } from 'react';
import { searchImplementation } from '../redux/covidcases';

import CasesContainer from './CasesContainer';
import Header from './Header';
import Menu from './Menu';

const Home = () => {
  const { all } = useSelector((state) => state.cases);

  const [state, setState] = useState({
    value: '',
    filteredCases: [],
  });

  const handleChange = (e) => {
    const str = e.target.value;
    const searchStr = str.charAt(0).toUpperCase() + str.slice(1);
    setState({
      value: e.target.value,
      filteredCases: [...searchImplementation(all, searchStr)],
    });
  };

  return (
    <>
      <Header />
      <Menu />
      {
        state.filteredCases.length > 0 ? (
          <CasesContainer
            cases={state.filteredCases}
            value={state.value}
            handleChange={handleChange}
          />
        ) : (
          <CasesContainer
            cases={all}
            value={state.value}
            handleChange={handleChange}
          />
        )
      }
    </>
  );
};

export default Home;
