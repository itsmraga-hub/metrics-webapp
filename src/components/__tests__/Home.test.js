import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';

import Home from '../Home';

it('<Home /> Renders Properly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
