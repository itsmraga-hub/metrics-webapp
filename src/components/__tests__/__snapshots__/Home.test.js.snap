// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Renders Properly 1`] = `
Array [
  <header
    className="Header"
  >
    <svg
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
      />
    </svg>
    <p>
      All time stats
    </p>
    <div
      className="HeaderDiv"
    >
      <svg
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M192 0C139 0 96 43 96 96V256c0 53 43 96 96 96s96-43 96-96V96c0-53-43-96-96-96zM64 216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 89.1 66.2 162.7 152 174.4V464H120c-13.3 0-24 10.7-24 24s10.7 24 24 24h72 72c13.3 0 24-10.7 24-24s-10.7-24-24-24H216V430.4c85.8-11.7 152-85.3 152-174.4V216c0-13.3-10.7-24-24-24s-24 10.7-24 24v40c0 70.7-57.3 128-128 128s-128-57.3-128-128V216z"
        />
      </svg>
      <svg
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
        />
      </svg>
    </div>
  </header>,
  <section
    className="Menu"
  >
    <h2>
      All Time Stats
    </h2>
    <div>
      <p>
        0
        <span>
          Total confirmed cases
        </span>
      </p>
      <p>
        0
        <span>
          Population
        </span>
      </p>
      <p>
        0
        <span>
          Total Deaths
        </span>
      </p>
    </div>
  </section>,
  <section
    className="Section"
  >
    <div
      className="SearchContainer"
    >
      <input
        className="InputField"
        id="searchbar"
        name="search"
        onChange={[Function]}
        placeholder="Filter/search by country"
        value=""
      />
    </div>
    <ul
      className="Container"
    />
  </section>,
]
`;
